pkg/common/interceptors.go

```diff
@@ -0,0 +1,165 @@
+package common
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/container-storage-interface/spec/lib/go/csi"
+	"github.com/kubernetes-csi/csi-lib-utils/protosanitizer"
+	"github.com/kubernetes-sigs/alibaba-cloud-csi-driver/pkg/metric"
+	"github.com/kubernetes-sigs/alibaba-cloud-csi-driver/pkg/options"
+	"github.com/prometheus/client_golang/prometheus"
+	log "github.com/sirupsen/logrus"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const (
+	podNameKey          = "csi.storage.k8s.io/pod.name"
+	podNamespaceKey     = "csi.storage.k8s.io/pod.namespace"
+	ControllerPublish   = "ControllerPublishVolume"
+	ControllerUnpublish = "ControllerUnpublishVolume"
+	NodeStage           = "NodeStageVolume"
+	NodeUnstage         = "NodeUnstageVolume"
+	NodePublish         = "NodePublishVolume"
+	NodeUnpublish       = "NodeUnpublishVolume"
+)
+
+var client *kubernetes.Clientset
+
+type reqWithVolumeId interface {
+	GetVolumeId() string
+}
+
+func init() {
+	client = getK8sClient()
+}
+
+func getK8sClient() *kubernetes.Clientset {
+	config, err := clientcmd.BuildConfigFromFlags(options.MasterURL, options.Kubeconfig)
+	if err != nil {
+		log.Errorf("interceptors: initializing kubernetes config failed: %s", err)
+		return nil
+	}
+	client, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		log.Errorf("interceptors: initializing kubernetes clientset failed: %s", err)
+		return nil
+	}
+	return client
+}
+
+func logGRPC(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+	log.Infof("GRPC call: %s", info.FullMethod)
+	log.Debugf("GRPC request: %s", protosanitizer.StripSecrets(req))
+	resp, err := handler(ctx, req)
+	if err != nil {
+		log.Errorf("GRPC error: %v", err)
+	} else {
+		log.Debugf("GRPC response: %s", protosanitizer.StripSecrets(resp))
+	}
+	return resp, err
+}
+
+func instrumentGRPC(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler, driverType string) (interface{}, error) {
+	if _, ok := req.(reqWithVolumeId); !ok {
+		return handler(ctx, req)
+	}
+
+	volumeId := req.(reqWithVolumeId).GetVolumeId()
+	method := getMethodName(info.FullMethod)
+	switch method {
+	case ControllerPublish, ControllerUnpublish, NodeStage, NodeUnstage, NodeUnpublish:
+		start := time.Now()
+		resp, err := handler(ctx, req)
+		execTime := time.Since(start).Milliseconds()
```

Comment:
```suggestion
		execTime := time.Since(start).Seconds()
```
Use seconds as it already returns a float64.